### PR TITLE
Add embedding tutorial

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Add embedding model tutorial

## How has it been tested?

* Docs: [Embedding Model for Satellite Images with Torchvision’s ResNet - LightlyTrain documentation.pdf](https://github.com/user-attachments/files/20190084/Embedding.Model.for.Satellite.Images.with.Torchvision.s.ResNet.-.LightlyTrain.documentation.pdf)
* Colab: https://colab.research.google.com/github/lightly-ai/lightly-train/blob/guarin-trn-316-add-embedding-tutorial/examples/notebooks/torchvision_embedding_model.ipynb
  * Images in the colab will only be visible once this branch is merged to main.
  * Ran everything through with training epochs set to 1

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/tutorials/embedding/index.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+plt.title("UMAP of LightlyTrain Embeddings")
+plt.show()
+```
+
+```{figure} /_static/images/tutorials/embedding/umap_lightly_train.jpg
+Visualization of the learned embeddings projected into 2D space with UMAP.
+```
+
+We can see that the embeddings are nicely separated into well-defined clusters. This
+means that the model has learned meaningful representations. Such visualizations are
```
Comment (stegmuel) [inline docs/source/tutorials/embedding/index.md]: Until the clusters are colored I don't think we can deduce anything with regards to whether or not the learned representations are meaningful or not.

File: docs/source/tutorials/embedding/index.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+plt.scatter(embedding_2d[:, 0], embedding_2d[:, 1], s=5, c=color, cmap="tab20")
+plt.title("UMAP of LightlyTrain Embeddings Colored by Class")
+plt.show()
+```
+
+```{figure} /_static/images/tutorials/embedding/umap_lightly_train_colored.jpg
+Embeddings colored by ground truth class labels.
+```
+
+The embeddings are well separated by class with few outliers. LightlyTrain has learned
```
Comment (stegmuel) [inline docs/source/tutorials/embedding/index.md]:
```suggestion
The embeddings are well separated by class with few outliers. The LightlyTrain model has learned
```

File: docs/source/tutorials/embedding/index.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+plt.savefig("umap_lightly_train_imagenet_colored.jpg", bbox_inches="tight")
+plt.show()
+```
+-->
+
+We can see that the clusters from the LightlyTrain embeddings are much more compact
+and have fewer overlaps. This means that the model has learned better representations
+and will make fewer mistakes for embedding-based tasks like image retrieval or
+clustering. This highlights how training an embedding model on the target dataset can
+improve the embedding quality significantly compared to using an off-the-shelf
```
Comment (stegmuel) [inline docs/source/tutorials/embedding/index.md]:
```suggestion
improve the embeddings quality significantly compared to using an off-the-shelf
```

Comment (stegmuel) [reply docs/source/tutorials/embedding/index.md]: Not super comfortable with talking about "significant quality" improvement when the only assessment is visualization of the embeddings after dimensionality reduction.

File: docs/source/tutorials/embedding/index.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+embedding model.
+
+## Conclusion
+
+In this tutorial we have learned how to train an embedding model using unlabeled data
+with LightlyTrain. We have also seen how to visualize the embeddings with UMAP and
+color them according to class labels. The visualizations show that the model has learned
+strong embeddings that capture the information of the images well and group similar
+images together. This is a great starting point for any embedding-based task such as
+image retrieval, clustering, outlier detection or dataset curation.
```
Comment (stegmuel) [inline docs/source/tutorials/embedding/index.md]: Probably also for finetuning, no?

**Review by stegmuel — COMMENTED** [review]
The tutorial looks great and is easy to follow.
I left some comments.

**Review by stegmuel — APPROVED** [review]
LGTM!
